gen/src/main/java/com/softlayer/api/gen/ClassWriter.java

```diff
@@ -377,8 +377,14 @@ public ClassWriter emitType() throws IOException {
         
         // Now the service
         if (!type.meta.noservice) {
-            if (type.meta.properties.containsKey("id")) {
-                if (type.meta.properties.containsKey("globalIdentifier")) {
+
+            Boolean containsId = type.meta.properties.containsKey("id")
+                    || type.baseMeta.properties.containsKey("id");
```

Comment:
Yeah, I was trying to see how that might work, but we don't have a way to recurse easily in ClassWriter because the TypeClass contains the base Meta.Type instead of the Meta.Type containing the base Meta.Type. You have to either keep a reference to the Meta object in the ClassWriter to look up by name, or provide a reference to the base Meta.Type on a Meta.Type, but that would require an iteration over all of the types to add the references, because you aren't guaranteed that a parent type will be in the metadata before a child type..

I might be missing something though.